src/main/java/org/opensrp/repository/postgres/SettingRepositoryImpl.java

```diff
@@ -520,7 +538,60 @@ private void checkWhetherMetadataExistsBeforeSave(SettingConfiguration entity, L
 				settingsMetadataList.add(metadata);
 			}
 		}
-			settingMetadataMapper.insertMany(settingsMetadataList);
+		insertSettingMetadata(settingsMetadataList);
+	}
+
+	private void insertSettingMetadata(List<SettingsMetadata> settingsMetadataList) throws SQLException {
+		String insertSettingMetadata = "INSERT INTO core.settings_metadata ( settings_id, document_id, identifier, "
+				+ "server_version, team, team_id, provider_id, location_id, uuid, json, setting_type, setting_value, setting_key, setting_description, setting_label, inherited_from) VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)";
+
+		Connection connection = null;
+
+		try {
+			connection = DataSourceUtils.getConnection(openSRPDataSource);
+			PreparedStatement preparedStatement = connection.prepareStatement(insertSettingMetadata);
+			for (SettingsMetadata settingsMetadata : settingsMetadataList) {
+
+				ObjectMapper objectMapper = new ObjectMapper();
+				String json = null;
+				try {
+					json = objectMapper.writeValueAsString(settingsMetadata.getJson());
+				}
+				catch (JsonProcessingException e) {
```

Comment:
Use Logger.error instead of `e.printStackTrace();`

Also if there is an error shouldn't code below be skipped? i.e there is a `continue`

Reply:
This means that a specific setting won't be saved.

Reply:
yes, if an exception occurs this means the saved metadata will save invalid or null value